builder/pipeline.cc

```diff
@@ -886,29 +886,76 @@ class pipeline_builder {
       for (const auto& i : f->inputs()) {
         const auto& input = i.buffer;
 
-        if (!input->producer()) {
-          continue;
-        }
-
-        if (!f->impl() && !f->padding()) {
+        if ((!f->impl() && !f->padding()) || (input->constant())) {
           // Collect all buffers which are inputs to the copy
           // and the outputs of the copy as their dependencies.
           copy_inputs_.insert(input->sym());
           for (const func::output& o : f->outputs()) {
             const buffer_expr_ptr& b = o.buffer;
-            if (output_syms_.count(b->sym())) continue;
+            if (!input->constant() && output_syms_.count(b->sym())) continue;
             copy_deps_[input->sym()].push_back(b->sym());
           }
         }
 
-        allocation_info_[input->sym()]->deps_count++;
+        if (allocation_info_[input->sym()]) {
+          allocation_info_[input->sym()]->deps_count++;
+        }
+      }
+    }
+  }
+
+  bool has_all_allocations(buffer_expr_ptr candidate, std::set<var>& allocations) {
+    return std::all_of(copy_deps_[candidate->sym()].begin(), copy_deps_[candidate->sym()].end(),
```

Comment:
In this case, `std::all_of` isn't doing much, because you can just return from inside a loop.

Reply:
I guess makes less sense when I moved it into a helper functions.